ios/RCTTWVideoModule.m

```diff
@@ -640,6 +641,13 @@ -(TVIVideoBandwidthProfileOptions*)prepareBandwidthProfile:(NSDictionary *)bandw
   [self clearCameraInstance];
   [self stopLocalAudio];
   [self.room disconnect];
+
+  self.camera = nil;
+  self.localVideoTrack = nil;
+  self.localAudioTrack = nil;
+  self.localDataTrack = nil;
+  self.localParticipant = nil;
+  self.room = nil;
```

Comment:
It feels that we already clean some of these properties when calling the methods on top. 
For instance, `localAudioTrack`, `localDataTrack`, and `room`  are already cleared on `stopLocalAudio`. 
We could also call `stopLocalVideo` which would also clear `localVideoTrack`. Maybe we can remove those?

For my understanding, in the upstream they only call `[self.room disconnect]` and the `[self clearCameraInstance]`. Why do we clean up everything else in our fork? Is it just to be sure we have a clean slate when a new call is started?

Reply:
This is a failsafe added to ensure that Twilio's "internal media factory" is definitely cleaned up (based on advise [here](https://github.com/twilio/video-quickstart-ios/issues/230#issuecomment-367467752)).

Without these additions, Twilio throws an exception when switching from a regular call to a custom audio call. I guess conditional custom audio isn't a common use case, so this bug hasn't been caught upstream.

You're right, we could just identify the specific object that's not cleaned up. But I think it's safer to nil everything here?

Reply:
Clear, yeps I don't mind cleaning everything just to be sure.
Should we have a separate method for it -similar to android NDExtra- to clearly show that this is something on us?

I like the fact that we use ND as a pre-fix, helping us later identify without checking upstream diffs whats ours and whats theirs. 

I leave it up to you guys since you are the ones working on it! Whatever you prefer.

Reply:
Cool!

> Should we have a separate method for it -similar to android NDExtra- to clearly show that this is something on us?

I don't think so. I think in this instance these changes should probably be included upstream (maybe we should make a PR?).

But I really like the idea of identifying ND additions. Would a comment like `\\ ND Ensure the internal media factory is cleaned up to avoid an exception when switching to custom audio` do for now?